frontend/packages/dev-console/src/components/topology2/components/nodes/BaseNode.tsx

```diff
@@ -72,21 +82,23 @@ const BaseNode: React.FC<BaseNodeProps> = ({
   const refs = useCombineRefs<SVGEllipseElement>(hoverRef, dragNodeRef);
 
   React.useLayoutEffect(() => {
-    if (hover) {
-      onShowCreateConnector && onShowCreateConnector();
-    } else {
-      onHideCreateConnector && onHideCreateConnector();
+    if (editAccess) {
+      if (hover) {
+        onShowCreateConnector && onShowCreateConnector();
+      } else {
+        onHideCreateConnector && onHideCreateConnector();
+      }
     }
-  }, [hover, onShowCreateConnector, onHideCreateConnector]);
+  }, [hover, onShowCreateConnector, onHideCreateConnector, editAccess]);
 
   return (
     <g className="odc2-base-node">
       <NodeShadows />
       <g
         data-test-id="base-node-handler"
         onClick={onSelect}
-        onContextMenu={onContextMenu}
-        ref={refs}
+        {...(editAccess ? { onContextMenu } : {})}
+        {...(editAccess ? { ref: refs } : {})}
```

Comment:
`refs` provides drag to move behavior AND regroup behavior.
The regroup behavior needs to suppressed but not the drag.

Look at `componentUtils#nodeDragSourceSpec`. This is where the `REGROUP_OPERATION` is installed.
We may need to move to hooks instead of HOCs to gain better control of the behaviors.